### PR TITLE
Hacktoberfest: Project Euler Solution Problem 50

### PR DESCRIPTION
**Pull request by dhvl-nyk**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by NSachida — APPROVED** [review]
Answer: 997651

**Review by 53jk1 — APPROVED** [review]

File: project_euler/problem_050/sol1.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+The longest sum of consecutive primes below one-thousand that adds to a prime,
+contains 21 terms, and is equal to 953.
+
+Which prime, below one-million, can be written as the sum of the most consecutive
+primes?
+"""
+
+
+def binary_search(arr, left, right, element):
```
Comment (poyea) [inline project_euler/problem_050/sol1.py:20]: Please add some doctest.

File: project_euler/problem_050/sol1.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            return mid
+        elif arr[mid] > element:
+            return binary_search(arr, left, mid - 1, element)
+        else:
+            return binary_search(arr, mid + 1, right, element)
+    else:
+        return -1
+
+
+def sieve(limit):
```
Comment (poyea) [inline project_euler/problem_050/sol1.py:33]: Please add some doctest is possible

**Comment by poyea** [comment]
Maybe you'll have to add some answer to the code too. If you're unsure, please read again the https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md

**Comment by dhruvmanila** [comment]
By answer if you are referring to the line `return answer` then that is just a way of telling the user to return the answer to the problem, they can have whatever variable name they want.

**Pull request closed without merging**